llama_stack/distribution/server/quota.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+import json
+from datetime import datetime, timezone
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+from llama_stack.log import get_logger
+from llama_stack.providers.utils.kvstore.api import KVStore
+from llama_stack.providers.utils.kvstore.config import KVStoreConfig, SqliteKVStoreConfig
+from llama_stack.providers.utils.kvstore.kvstore import kvstore_impl
+
+logger = get_logger(name=__name__, category="quota")
+
+
+class QuotaMiddleware:
+    """
+    ASGI middleware enforcing per client daily request quotas.
+
+    Expects Authorization: Bearer <client_id> header.
+    Tracks counts in a KV store (SQLite); returns HTTP 429 when limit is exceeded.
+    """
+
+    def __init__(
+        self,
+        app: ASGIApp,
+        kv_config: KVStoreConfig | None = None,
+        default_requests_per_day: int = 1000,
+        window_seconds: int = 86400,
+    ):
+        self.app = app
+        # if no config passed, default to on disk SQLite
```

Comment:
I believe this contradicts the above statement that if no config if passed quota are disabled. When I look at the caller code `kv_config` cannot be None. Since we initialize `        kv_config = SqliteKVStoreConfig(db_path=quota_conf.db_path)`.

The `or` seems superfluous here.

Reply:
Nice catch, I dropped it!